src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -134,7 +165,7 @@ def _compute_metrics(self, sorting_analyzer, unit_ids=None, verbose=False, **job
 
         import pandas as pd
 
-        metrics = pd.DataFrame(index=unit_ids)
+        metrics = pd.DataFrame(index=sorting_analyzer.sorting.unit_ids)
```

Comment:
```suggestion
        metrics = pd.DataFrame(index=sorting_analyzer.unit_ids)
```
A small suggestion, but let's use the strengths of the `SortingAnalyzer` :)